app/models/distribution.rb

```diff
@@ -9,6 +9,45 @@ class Distribution < BaseModel
   has_many :audio_version_templates, through: :distribution_templates
   serialize :properties, HashSerializer
 
+  def self.check_published!(data = nil)
+    data ||= {}
+    from_secs = (data[:from_secs] || '60').to_i
+    to_secs = (data[:to_secs] || '3600').to_i
+    story_id = nil
+    dist_id = nil
+
+    recently_published_stories(from_secs, to_secs).each do |story|
+      story_id = story.id
+      dist_id = nil
+      story.distributions.each do |dist|
+        dist_id = dist.id
+        if !dist.distributed?
+          notice_message("Story #{story_id} not distributed: #{dist.distribution.url}")
+          dist.distribute!
+        end
+
+        if !dist.published?
+          notice_message("Story #{story_id} distribution not published: #{dist.url}")
+          dist.publish!
+        end
+      end
+    end
+  rescue StandardError => e
+    NewRelic::Agent.notice_error(e, custom_params: { story: story_id, distribution: dist_id })
```

Comment:
would it be helpful to have the time range in the NR log too?

Reply:
the range shouldn't really change, so I didn't think it would help much